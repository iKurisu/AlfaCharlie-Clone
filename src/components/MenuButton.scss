.menu-button {
  width: 60px;
  height: 60px;
  padding: 20px;
  position: fixed;
  top: 20px;
  right: 33px;
  cursor: pointer;
}
