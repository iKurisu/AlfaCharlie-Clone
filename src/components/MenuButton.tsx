import React from "react";
import Hamburger from "./menu_button/Hamburger";
import Cross from "./menu_button/Cross";
import "./MenuButton.scss";

interface Props {
  menuIsOpen: boolean;
  toggle: () => void;
}

const MenuButton = ({ menuIsOpen, toggle }: Props): JSX.Element => (
  <div className={`menu-button${menuIsOpen ? " open" : ""}`} onClick={toggle}>
    <Hamburger />
    <Cross />
  </div>
);

export default MenuButton;
