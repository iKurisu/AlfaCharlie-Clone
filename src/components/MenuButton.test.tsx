import React, { useState } from "react";
import renderer from "react-test-renderer";
import Enzyme, { shallow, mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import MenuButton from "./MenuButton";

Enzyme.configure({ adapter: new Adapter() });

describe("menu button", (): void => {
  it("renders correctly", (): void => {
    const toggle = jest.fn();
    const button = renderer.create(
      <MenuButton menuIsOpen={false} toggle={toggle} />
    );

    expect(button).toMatchSnapshot();

    const button2 = renderer.create(
      <MenuButton menuIsOpen={true} toggle={toggle} />
    );

    expect(button2).toMatchSnapshot();
  });

  it("calls the event handler", (): void => {
    const toggle = jest.fn();
    const button = shallow(<MenuButton menuIsOpen={false} toggle={toggle} />);

    button.find("div").simulate("click");

    expect(toggle.mock.calls.length).toBe(1);
  });

  it("updates the class", (): void => {
    const Wrapper = (): JSX.Element => {
      const [menuIsOpen, openMenu] = useState<boolean>(false);

      const toggleMenu = (): void => openMenu(!menuIsOpen);

      return <MenuButton menuIsOpen={menuIsOpen} toggle={toggleMenu} />;
    };

    const wrapper = mount(<Wrapper />);
    wrapper.find(".menu-button").simulate("click");

    expect(wrapper.find(".menu-button open")).toBeTruthy();
  });
});
