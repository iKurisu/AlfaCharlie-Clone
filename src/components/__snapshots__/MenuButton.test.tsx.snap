// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`menu button renders correctly 1`] = `
<div
  className="menu-button"
  onClick={[MockFunction]}
>
  <svg
    className="menu-hamburger"
    viewBox="0 0 72 72"
    x="0"
    y="0"
  >
    <line
      x1="2"
      x2="69"
      y1="9"
      y2="9"
    />
    <line
      x1="2"
      x2="69"
      y1="35"
      y2="35"
    />
    <line
      x1="2"
      x2="69"
      y1="61"
      y2="61"
    />
  </svg>
  <svg
    className="menu-cross"
  />
</div>
`;

exports[`menu button renders correctly 2`] = `
<div
  className="menu-button open"
  onClick={[MockFunction]}
>
  <svg
    className="menu-hamburger"
    viewBox="0 0 72 72"
    x="0"
    y="0"
  >
    <line
      x1="2"
      x2="69"
      y1="9"
      y2="9"
    />
    <line
      x1="2"
      x2="69"
      y1="35"
      y2="35"
    />
    <line
      x1="2"
      x2="69"
      y1="61"
      y2="61"
    />
  </svg>
  <svg
    className="menu-cross"
  />
</div>
`;
